### PR TITLE
Allow private key to be saved as file on host

### PR DESCRIPTION
**Pull request by phelpsdb**
Specify either private key directly or as an absolute path to a key file on the host.

Tested on stage dn5

### CODE REVIEW

File: pkg/conf/model.go [hunk 1]
```diff
@@ -21,9 +21,24 @@ func NewContextConfig() *ContextConfig {
...
 // base structure that all server types need
 type NodeConfig struct {
+	// *** Required fields ***
+
 	Type NodeType `yaml:"type"`
+	// The delegate owner key can either be directly specified or an
+	// absolute path to a file on the HOST machine (not your local machine) containing the key
+	PrivateKey    string `yaml:"privateKey"`
+	Wallet        string `yaml:"wallet"`
+	RewardsWallet string `yaml:"rewardsWallet"`
```
Comment (alecsavvy) [inline pkg/conf/model.go:31]: i like this naming a lot, more clear than the "delegate" terminology imo

File: pkg/conf/model.go [hunk 2]
```diff
@@ -21,9 +21,24 @@ func NewContextConfig() *ContextConfig {
...
+	// One of "current", "prerelease", or an audius-docker-compose git branch (for dev purposes)
+	// "current" corresponds to main adc branch
+	// "prelease" corresponds to stage
+	// defaults to "current" behavior if omitted
+	Version string `yaml:"version,omitempty"`
+
+	// *** Optional fields ***
 
-	// (Optional) Specify non-standard ports for http traffic
+	EthWallet string `yaml:"ethWallet,omitempty"`
```
Comment (alecsavvy) [inline pkg/conf/model.go:40]: what's the significance of EthWallet over the other two wallets? aren't all three technically ethereum wallets?

File: pkg/conf/model.go [hunk 3]
```diff
@@ -32,25 +47,14 @@ type NodeConfig struct {
...
-	RewardsWallet string `yaml:"rewardsWallet"`
-	EthWallet     string `yaml:"ethWallet,omitempty"`
-
-	// (Optional) Remote content storage
+	// Remote content storage
 	StorageUrl         string `yaml:"storageUrl,omitempty"`
 	StorageCredentials string `yaml:"storageCredentials,omitempty"`
 
-	// (Optional) Postgres db url for remote db and/or custom password
+	// Postgres db url for remote db and/or custom password
```
Comment (alecsavvy) [inline pkg/conf/model.go:54]: nice, I think there may be some SPs using hosted dbs. blockdaemon nodes for example only have 91gigs on disk which doesn't make sense considering the size of the discovery db + acdc

File: pkg/orchestration/docker.go [hunk 4]
```diff
@@ -195,8 +194,16 @@ func runNode(
...
+		privateKey, err := NormalizedPrivateKey(host, config.PrivateKey)
+		if err != nil {
+			return logger.Error(err)
+		}
+		config.PrivateKey = privateKey
+	}
+	override := config.ToOverrideEnv(host, nconf)
 	// generate the override.env file locally
-	// WARNING: NOT THREAD SAFE
+	// WARNING: not thread safe due to constant filename
```
Comment (alecsavvy) [inline pkg/orchestration/docker.go:206]: thread safe in the way that multiple audius-ctl commands could be generating the same file?

File: pkg/conf/model.go (hunk 2 above)
Comment (phelpsdb) [reply pkg/conf/model.go:40]: I honestly don't know, the creator node just has a separate config for "ethOwnerWallet" and I don't know why or how it differs from "spOwnerWallet". Could just be more legacy cruft that I'd be happy to get rid of.

Comment (alecsavvy) [reply pkg/conf/model.go:40]: gotcha, it's probably the same but we can fix that later

**Review by alecsavvy — APPROVED** [review]

**Pull request merged**